Eugen

theugen opened the pull request:
Nuova pull request (poi per questo we la pianto, promesso).
Codice decisamente più ordinato rispetto al vecchio, dà ancora un problema con le lunghezze di data$freq e data$peaks con differenza di un elemento per il dataset 20160129.16r... (gli altri nessun problema). Nei commits qualche commento in più.
Buon we

theugen committed:
Modifica nomi file

Per il ciclo generale da fare ho usato sub() con una regex per eliminare il .dat alla fine del nome die file.
Può essere utile spezzare in due ulteriori parti, per far sì che la prima (giorno di presa dati, aka "nome campagna")
possa eventualmente venire usata come nome per una cartella da creare automaticamente, l'altra come ora per poter
essere stampata sui grafici.
Lavoro appena iniziato, lo finisco appena possibile.

theugen committed:
Ciclo generale e path

Ho continuato le modifiche per il ciclo generale sui file cominciate ieri.
Ora il ciclo viene fatto da main.R, che importa il file e va a chiamare gli script
Correlation.R e Periodigram.R. Così dovrebbe essere più efficiente, con un'esecuzione
sola si fanno tutte le prese dati. (ovvero schiacciamo il bottone e andiamo a prendere
un caffè :P)
Il lavoro è ancora da finire, devo mettere a posto i nomi dei grafici con il percorso giusto
e un po' di roba, togliere un po' di comandi che vengono chiamati un po' di volte, e così
via, ma almeno è un inizio.

 On branch Eugen
 Changes to be committed:
	modified:   Correlation.R
	modified:   Periodigram.R
	modified:   main.R

theugen committed:
On branch Eugen
 Changes to be committed:
	modified:   Correlation.R

theugen committed:
Pulizia script e percorsi

Ho tolto il codice inutilizzato, ho sistemato la questione dei nomi dei file e ora viene
tutto chiamato dal main.R. Così mi sembra più leggibile...
Se si vuole far girare solo uno dei due script, commentare lo/gli script inutilizzati
nel main.
Nota: facendo il ciclo totale, si impalla sul dataset 20160129.16r, dice che

Error in (function (formula, data = NULL, subset = NULL, na.action = na.fail,  :
  variable lengths differ (found for 'data$freq')

Effettivamente la lunghezza di data$freq=36005, quella di data$peaks=36006,
mi sa che ci sta comunque sfuggendo qualcosa...

 On branch Eugen
 Changes to be committed:
	modified:   Correlation.R
	modified:   Periodigram.R
	modified:   functions.R
	modified:   main.R

algebrato commented:
Mi dispiace se questo week end non hon fatto nulla. Domani sera se riesco controllo tutto per bene

theugen commented:
Tranquillo!
Riguardo al problema della chiamata a dir.exist() in create_directory(), ho provato a guardare ma mi sa che la soluzione di includere il sorgente in un nostro script non può funzionare...
Ho provato a farmi sputare il sorgente da Rstudio e mi restituisce questo output

`> dir.exists
function (paths) 
.Internal(dir.exists(paths))
<bytecode: 0x3b4e768>
<environment: namespace:base>`

==> funzione interna, ovvero scritta in C... installo il pacchetto pryr per poter vedere il sorgente, mi dà questo link https://github.com/search?q=SEXP%20attribute_hidden%20do_direxists+repo:wch/r-source&type=Code , dove vedo che il sorgente è

```
SEXP attribute_hidden do_direxists(SEXP call, SEXP op, SEXP args, SEXP rho)
{
    SEXP fn, ans;

#ifdef Win32
    struct _stati64 sb;
#else
    struct stat sb;
#endif

    checkArity(op, args);
    fn = CAR(args);
    if (!isString(fn))
    error(_("invalid filename argument"));
    int n = LENGTH(fn);
    PROTECT(ans = allocVector(LGLSXP, n));
    for (int i = 0; i < n; i++) {
#ifdef Win32
    wchar_t *wfn = filenameToWchar(STRING_ELT(fn, i), TRUE);
    /* trailing \ is not valid on Windows except for the
       root directory on a drive, specified as "\", or "D:\",
       or "\\?\D:\", etc.  We remove it in other cases,
       to help those who think they're on Unix. */
    size_t len = wcslen(wfn);
    if (len) {
        wchar_t *p = wfn + (len - 1);
        if (len > 1 && (*p == L'/' || *p == L'\\') &&
        *(p-1) != L':') *p = 0;
    }
#else
    const char *efn = R_ExpandFileName(translateChar(STRING_ELT(fn, i)));
#endif
    if (STRING_ELT(fn, i) != NA_STRING &&
#ifdef Win32
        _wstati64(wfn, &sb)
#else
        /* Target not link */
        stat(efn, &sb)
#endif
        == 0) {
        LOGICAL(ans)[i] = (sb.st_mode & S_IFDIR) > 0;

    } else LOGICAL(ans)[i] = 0;
    }
    // copy names?
    UNPROTECT(1);
    return ans;
}
```

Mi sa che la soluzione è aggiornare R o, se tutti e 3 doveste aver problemi, trovare il modo di aggirare il problema... Anna, Chiara, se dopo aver fatto girare lo script functions.R provate a lanciare create_directory('./cartella_prova') avete problemi o vi gira tutto?

Buona serata a tutti!

theugen committed:
Re: Problema nell'antitrasformata

Ho provato a guardare il problema segnalato da Stefano nel suo
ultimo commit, "Dimensione Freq ok --- Problema nell'antitrasformata".
Ammetto che ho seguito un approccio molto grezzo, ho provato a guardare
se c'era qualche finestra che agli estremi del dominio fosse diversa
da zero, a differenza di quella di Hann.
Da wiki:
https://it.wikipedia.org/wiki/Funzione_finestra#Finestra_rettangolare
ho trovato una carrellata di funzioni finestra, alcune delle quali
potrebbero fare al caso nostro...
visto che avevamo a disposizione il pacchetto 'e1071', già usato
per la finestra di Hann, ho provato quella di Hamming.
Se devo essere sincero non mi ricordavo più in che dataset avevamo visto
"sparare" ai bordi, in quello che ho provato io era abbastanza accentuato
ma non ai livelli di quello che ho visto con Stefano oggi.
In ogni caso la finestra di Hamming sembra più docile agli estremi rispetto
a quella di Hann, fatemi sapere.
Un'altra finestra papabile potrebbe essere quella di Gauss, ma non ho la
più pallida idea di come si implementi (oh, volendo la si fa, ma potendo
evitare...)

Buona serata a tutti!

 On branch Eugen
 Changes to be committed:
	modified:   Periodigram.R

theugen committed:
Errata Corrige:
Scusate, avevo caricato una versione con i commenti sminchiati,
questa dovrebbe essere quella corretta.
Confermo che anche su altri dataset Hamming sembrerebbe ridurre
l'instabilità ai bordi, poi vedremo insieme.

 On branch Eugen
 Changes to be committed:
	modified:   Periodigram.R

algebrato commented:
Sì concordo. Non ti preoccupare un modo lo trovo per far runnare dir.exist. Appena riesco mi metto a pasticciare.
